### PR TITLE
Set up staticcheck linter for BFF

### PR DESCRIPTION
**Pull request by Griffin-Sullivan**
## Description
Linting the BFF folder with staticcheck and adding a step to run the linter in the BFF GitHub Action.

## How Has This Been Tested?
Make sure the action passes and tests pass.

## Merge criteria:

- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [x] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work.
- [x] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).

If you have UI changes

- [ ] The developer has added tests or explained why testing cannot be added.
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

File: .github/workflows/ui-bff-build.yml [hunk 1]
```diff
@@ -24,6 +24,11 @@ jobs:
         working-directory: clients/ui/bff
         run: make clean
 
+      - uses: dominikh/staticcheck-action@v1
```
Comment (ederign) [inline .github/workflows/ui-bff-build.yml:27]: @tarilabs is there a policy to add third part actions on kubeflow?

**Review by ederign — CHANGES_REQUESTED** [review]
LGTM to me and thanks for the cleanup! Just want to double check that with Matteo!

**Comment by Griffin-Sullivan** [comment]
I've decided to use a different linting tool for the BFF so I'm going to close this and open another.

**Pull request closed without merging**